fw/src/ui/settings.rs

```diff
@@ -0,0 +1,86 @@
+//! Settings page for application UI
+//!
+
+use rand_core::{CryptoRng, RngCore};
+
+use nanos_sdk::buttons::ButtonEvent;
+use nanos_ui::{
+    bagls::*,
+    layout::{Draw, Layout, Location, StringPlace},
+    screen_util,
+};
+
+use ledger_mob_core::{
+    apdu::tx::{FogId, FOG_IDS},
+    engine::{Driver, Engine},
+};
+
+use super::{clear_screen, UiResult};
+
+/// [Settings] page, at the moment this only provides Fog configuration
+#[derive(PartialEq, Clone, Debug)]
+pub struct Settings {
+    fog_id_index: usize,
+}
+
+impl Settings {
+    pub fn new(fog_id: FogId) -> Self {
+        let fog_id_index = fog_id as usize;
+        Self { fog_id_index }
+    }
+
+    pub fn update(&mut self, btn: &ButtonEvent) -> UiResult<FogId> {
+        match btn {
+            // Exit on both buttons pressed/released
+            ButtonEvent::BothButtonsRelease => UiResult::Exit(FOG_IDS[self.fog_id_index]),
+
+            // Otherwise move through fogs
+            ButtonEvent::LeftButtonRelease if self.fog_id_index > 0 => {
+                self.fog_id_index -= 1;
+                UiResult::Update
+            }
+            ButtonEvent::RightButtonRelease if self.fog_id_index < 4 => {
```

Comment:
Should this use `FOG_IDS.len()` instead of `4`?
And shouldn't it be `5` if using a hardcoded number? (None + 4 fogs)

Reply:
ohh yeah good call with `FOG_IDs.len()`... the logic is because we only want to be able to increment / scroll right when we're at `i < n` to give us a max of `i = n = 4`, same as we only decrement when `i > 0` so you can only scroll left to `i = 0`, which is correct but i'll have a think about whether it can be made more clear ^_^